appsmithorg/appsmith
# fix: added parsing logic in the array field

## Description
This PR fixes the UI freeze issue whenever an array of objects data is loaded into the JSON form.

NOTE: JSON form has auto generate mode enabled

#### PR fixes following issue(s)
Fixes #23622
>

#### Type of change
- Bug fix (non-breaking change which fixes an issue)
>
>
>
## Testing
>
#### How Has This Been Tested?

    - check if stringified defaultValue array generates correct number of array fields  
    - check if defaultValue array generates correct number of array fields
>
>
#### Test Plan
> Add Testsmith test cases links that relate to this PR
>
>
#### Issues raised during DP testing
> Link issues raised during DP testing for better visiblity and tracking (copy link from comments dropped on this PR)
>
>
>
## Checklist:
#### Dev activity


#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Test-plan-implementation#speedbreaker-features-to-consider-for-every-change) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans/_edit#areas-of-interest)
